be/src/vec/functions/cast_type_to_either.h

```diff
@@ -27,7 +27,7 @@ namespace doris::vectorized {
 class IDataType;
 
 template <typename... Ts, typename F>
-static bool cast_type_to_either(const IDataType* type, F&& f) {
+inline bool cast_type_to_either(const IDataType* type, F&& f) {
```

Comment:
here change from 'static' to 'inline' , does not affect the visible range of the function?

Reply:
```
inline marks the symbol WEAK, which hints linker to choose arbitary one of definition in object files
static marks the symbol LOCAL, which restricts the symbol in current translation unit, and the linker may keep multiple instances of (possibly different) definition.
```
https://gist.github.com/htfy96/50308afc11678d2e3766a36aa60d5f75

A header containing a static function is included by other files, which may generate an unused-template

Reply:
```For template functions, I'd say that using static, inline or no keyword produce the same result; in fact, in all cases the function will be inlined and no multiple definition error will be raised.
```
https://stackoverflow.com/questions/29495678/why-use-static-function-template